packages/app/obojobo-document-engine/src/scripts/viewer/components/nav/nav.js

```diff
@@ -0,0 +1,287 @@
+	// }
+
+	// focus() {
+	// 	Common.page.focus(this.selfRef)
+	// }
+
+	// onClickSkipNavigation() {
+	// 	FocusUtil.focusOnNavTarget()
+	// }
+
+	// renderLabel(label) {
+	// 	return label instanceof StyleableText ? <StyleableTextComponent text={label} /> : label
+	// }
+
+	// renderLinkButton(label, ariaLabel, isDisabled, refId = null) {
+	// 	return (
+	// 		<button ref={refId} aria-disabled={isDisabled} aria-label={ariaLabel}>
+	// 			{this.renderLabel(label)}
+	// 		</button>
+	// 	)
+	// }
+
+	// renderLink(index, isSelected, list, isItemDisabled, lockEl) {
+	// 	const item = list[index]
+	// 	const isFirstInList = !list[index - 1]
+	// 	const isLastInList = !list[index + 1]
+
+	// 	const className =
+	// 		'link' +
+	// 		isOrNot(isSelected, 'selected') +
+	// 		isOrNot(item.flags.complete, 'complete') +
+	// 		isOrNot(item.flags.correct, 'correct') +
+	// 		isOrNot(item.flags.assessment, 'assessment') +
+	// 		isOrNot(isFirstInList, 'first-in-list') +
+	// 		isOrNot(isLastInList, 'last-in-list')
+
+	// 	const labelText = getLabelTextFromLabel(item.label)
+	// 	let ariaLabel = labelText
+	// 	if (item.contentType) {
+	// 		ariaLabel = item.contentType + ' ' + ariaLabel
+	// 	}
+	// 	if (isSelected) {
+	// 		ariaLabel = 'Currently on ' + ariaLabel
+	// 	} else {
+	// 		ariaLabel = 'Go to ' + ariaLabel
+	// 	}
+
+	// 	return (
+	// 		<li key={index} onClick={this.onClick.bind(this, item)} className={className}>
+	// 			{this.renderLinkButton(item.label, ariaLabel, isItemDisabled, item.id)}
+	// 			{lockEl}
+	// 		</li>
+	// 	)
+	// }
+
+	// renderSubLink(index, isSelected, list, isItemDisabled, lockEl) {
+	// 	const item = list[index]
+	// 	const isLastInList = !list[index + 1]
+
+	// 	const className =
+	// 		'sub-link' +
+	// 		isOrNot(isSelected, 'selected') +
+	// 		isOrNot(item.flags.correct, 'correct') +
+	// 		isOrNot(isLastInList, 'last-in-list')
+
+	// 	const labelText = getLabelTextFromLabel(item.label)
+	// 	let ariaLabel = labelText
+	// 	ariaLabel = 'Jump to ' + labelText
+	// 	if (item.parent && item.parent.type && item.parent.type === 'link') {
+	// 		ariaLabel += ' inside ' + getLabelTextFromLabel(item.parent.label)
+	// 	}
+
+	// 	return (
+	// 		<li key={index} onClick={this.onClick.bind(this, item)} className={className}>
+	// 			{this.renderLinkButton(item.label, ariaLabel, isItemDisabled)}
+	// 			{lockEl}
+	// 		</li>
+	// 	)
+	// }
+
+	// renderHeading(index, item) {
+	// 	return (
+	// 		<li key={index} className={'heading is-not-selected'}>
+	// 			{item.label}
+	// 		</li>
+	// 	)
+	// }
+
+	getLockEl(isLocked) {
+		if (!isLocked) return null
+		return <div className="lock-icon" />
+	}
+
+	rendererNavSubItem() {
+		const { oboNodeList, adjList, navList, currentNavIndex, currFocusNode} = this.props
+		return adjList[navList[currentNavIndex]].map(childIndex => {
+			const node = oboNodeList[childIndex]
+			if(node.attributes.type !== 'ObojoboDraft.Chunks.Heading' && node.attributes.content.headingLevel !== 2) {
```

Comment:
(Some of these comments you may be aware of as this is a work in progress but I want to post them just in case.) This won't work in the final version as it's a hardcoded check for ObojoboDraft.Chunks.Heading, and we want developers to be able to create their own Heading content and have it work like ObojoboDraft.Chunks.Heading.

Reply:
Suggest having the node answer a question about itself (or just have a property). something like: `if(!node.isHeading && node.headingLevel !== 2)`

Perhaps that could be `Registry`'s job?  or some other inspector? `AllKnowingThing.isNodeAHeading(node)`